src/client/utils/hooks/useLockBodyScroll.ts

```diff
@@ -0,0 +1,20 @@
+import { useLayoutEffect } from 'react'
+
+type UseLockBodyScrollConfig = { lock: boolean }
+
+function useLockBodyScroll({ lock }: UseLockBodyScrollConfig = { lock: true }) {
+  useLayoutEffect(() => {
+    const previousOverflowStyle = window.getComputedStyle(document.body)
```

Comment:
This is a nice touch!

Just out of curiosity, why is it not enough doing?

```
    const previousOverflowStyle = document.body.style.overflow
```

Reply:
Because if you try to read a style property value of an element that hasn't been implicitly defined you may get as the return something that is actually not real. For example:

Consider a viewport which `width` is `200px`. If you render a `<div></div>` without any explicitly styling on that viewport and try to read it's `width` you'll get:

- `""` if you do it like this: `.style.width`
- `"200px"` if you do it like this: `window.getComputedStyle(element).width`, because a `div` is a block element that will take the whole `width` of its container. So it's **computed style** when it comes about `width` will be `200px`.

Reply:
Right, I understand that part. What I don't understand 💯 is why we don't just reset to the previous value? If there wasn't any previous value we just reset to "nothing" and let the browser recalculate what the computed value should be, no?

If we calculate the style before and store it we will add a new style to the element when we run the clean-up function. Seems better to delegate back to the browser.

Reply:
Haha okey, I'll try to be more clear:

```
const useLockBodyScroll = ({ lock = true }: UseLockBodyScrollConfig) => {
  useLayoutEffect(() => {
   const previousOverflowStyle = document.body.style.overflow

    if (lock) {
      document.body.style.overflow = 'hidden'
    }

    return () => {
      document.body.style.overflow = previousOverflowStyle
    }
  }, [lock])
}
```

This is a nit pick just to be clear. I don't really see a clear issue with your solution. Rather that this ⬆️ would be slightly simpler.

Reply:
Got it. 🤔 It seems it also works. This implementation is heavily based on [this](https://usehooks.com/useLockBodyScroll/) that's why the `getComputedStyle`. I couldn't see an edge case either just added the `getComputedStyle` because, you know, maybe the author did 😄

But, since it's working I gonna change to the proposed one.

Reply:
Haha with that reference I would probably also just go with the "computed style" solution 😅 so whatever feels best for you -- but great that we point to a reference!

Reply:
![](https://media2.giphy.com/media/143vPc6b08locw/giphy.gif)

😆 😆 😆 😆